### DISCUSSION TITLE
How to know when media is processed?

### DISCUSSION THREAD

**Discussion by AlejandroAkbal** [spatie/laravel-medialibrary#2915] [Q&A] [answered]
I'm building an image board.
When a user creates a post and uploads some media, it will get processed in a Redis queue.

The post gets published immediately, but because the media has not processed yet, the post will have broken (missing) media URLs.
What I want to do is: check if the post has all the media ready before publishing. Or publishing when the post is ready.

Is there any way to check if the media has been already processed? Is there an event or something similar?

**Comment by PaolaRuby** [comment]
Use sync instead of queue for posts

**Comment by AlejandroAkbal** [comment, marked as answer]
I ended up adding a column to the “Media” migration called “status”. Which is updated when all media processing is done.
